core/src/main/scala/org/apache/spark/sql/rapids/tool/qualification/QualificationAppInfo.scala

```diff
@@ -668,21 +676,19 @@ class QualificationAppInfo(
         None
       }
 
-      // get the ratio based on the Task durations that we will use for wall clock durations
-      // totalTransitionTime is the overhead time for ColumnarToRow/RowToColumnar transitions
-      // which impacts the GPU ratio.
-      val estimatedGPURatio = if (sqlDataframeTaskDuration > 0) {
-        supportedSQLTaskDuration.toDouble / (
-          sqlDataframeTaskDuration.toDouble + totalTransitionsTime.toDouble)
+      val (sqlDfWallDuration, supportedDfWallDuration) = getWallClockDurations(
+        appDuration, jobOverheadTime, sqlDataframeTaskDuration, supportedSQLTaskDuration)
+
+      // get the ratio based on the wallclock durations of supportedSqlWallClock duration to
+      // total sqlWallclock duration
+      val estimatedGPURatio = if (supportedDfWallDuration > 0) {
```

Comment:
this is a bit confusing because in side of the getWallClockDurations you already calculated ratios and now you calculate them again.  I kind of expected that to happen once inside of the calculateEstimatedInfoSummary.  I need to look at that closer.

Reply:
resolving this comment since commented below